### DISCUSSION TITLE
The client does not trust the proxy's certificate

### DISCUSSION THREAD

**Discussion by JamesPierce82** [mitmproxy/mitmproxy#5307] [Intercept Everything!]
As stated in the Topic, I am receiving an error for a specific domain stating:
"Client TLS handshake failed. The client does not trust the proxy's certificate for DOMAIN (tlsvs1 alert unkown ca)".

This error only appears to occur on specific domains for the app it is being generated from. Is this an example of Certificate pinning? Or have I perhaps configured something wrong along the way?

Looking for some direction on how to resolve the error as the data I am hunting seems to be being sent when that error is triggering.

**Comment by mhils** [comment]
Yes, this very much looks like certificate pinning. The client actively communicates with a TLS alert that it does not trust the cert.

If you're on Android and it's your own application, you need to add user-added CAs to your network configuration manifest.

**Reply by catscarlet** [reply]
I got same issue today. 
`Client TLS handshake failed. The client does not trust the proxy's certificate for api.somedomain.com (OpenSSL Error([('SSL routines', '', 'ssl/tls alert certificate unknown')]))`
The other apps are fine and can be proxied but only one app reported this one, which was the one I needed to analyze the traffic.
Both on iOS and Android. Wireguard mode.
I also have Magisk mod installed on Android. I can see the cert is in both user and sys.

What should I try next?

**Comment by JamesPierce82** [comment]
ah, that's what I was worried about. Thank you very much for the quick reply!

Unfortunately as the app I am working with is not my own, It looks like I will need to attempt some additional work using apk-mitm or objection as suggested in the Certificates documentation. 

Disappointing that i need to do more work, but exciting to be so close! By chance do you have any recommendations or suggestions on any "gatchas" to avoid during this next part?

**Reply by jorcelinojunior** [reply]
Have you made any progress? I'm having the same problem on android :(

**Reply by kevin0t** [reply]
Maybe Try running the app on an emulator with Android 7 or below or use root on it. By root method you could use magisk modules like lsposed -sslunpinning or [JustTrustMe](https://github.com/kasnder/JustTrustMe).

**Comment by ZaptorZap** [comment]
Something loosely related, but I was curious to peek into traffic coming from the _Nintendo Switch_ instead. Ran into the same issue as the title, although admittedly, I didn't and _certainly couldn't_ install any CA certificates. The connection test succeeded though, and I could peer into the HTTP `GET` that makes: address and all. All I've done was enable proxying on the Switch's internet settings and set the host as my computer. No homebrew involved here.

I escalated to trying to transparently proxy, but I just keep running into #3139 despite the fact that _my machine_ (Debian Bullseye running `./mitmproxy`) **shouldn't** even be proxying. mitmproxy quickly runs out of leeway and throws exceptions relating to too many open files, and the Switch fails at even testing its own connection. Really unsure of why this is happening. I'm quick to blame the [iptables listed on the docs](https://docs.mitmproxy.org/stable/howto-transparent/#3-create-an-iptables-ruleset-that-redirects-the-desired-traffic-to-mitmproxy), but I'll admit I have zero expertise in what I'm doing here.

**Comment by jidanni** [comment]
When making such messages, the app should be as helpful as possible.

> The client does not trust the proxy's certificate 

Which client. Be specific.

Which proxy. Be specific.

Note the app might be running under several layers of other apps. So be sure it says clearly what things it is talking about.

**Comment by selfstduy** [comment]
How to solve the hot problem

**Comment by bengabp** [comment]
pls how to solve this issue, im still getting the same message for linkedin for youtube

**Comment by strarsis** [comment]
The client must trust the CA cert used by `mitmproxy`, so you have to add the CA cert to the client trust store (Note: Firefox uses its own by default, independent from system store, there is an option to make it (also) use the system trust store, btw.). 

Also note that when `mitmproxy` is run as root (for using privileged ports as `443`/`80`), the initially, by default automatically generated CA certificates are not those in the normal user folder (when `mitproxy` was previously ran as normal user), but in root home directory (`/root/`), and those are randomly generated, so for root those differ to normal user.
